### PR TITLE
Fixed rubocop and foodcritic and added necessary files

### PR DESCRIPTION
**Pull request by cthalmann**
Chefspec tests need to be added to osl-varnish. In order to pass rake, this cookbook has to pass rubocop and foodcritic before even getting to chefspec. This PR fixes rubocop and foodcritic errors. I also used chef generate in order to add necessary files, such as .kitchen.yml and the Rakefile.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -4,3 +4,18 @@ AllCops:
...
+Metrics/LineLength:
+  Max: 120
+
+Style/IfUnlessModifier:
+  MaxLineLength: 120
+
+Style/WhileUntilModifier:
+  MaxLineLength: 120
+
+Metrics/BlockLength:
```
Comment (fahlmant) [inline .rubocop.yml:20]: Not needed now

File: spec/default_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+describe 'osl-varnish::default' do
+  ALL_PLATFORMS.each do |p|
+    context "#{p[:platform]} #{p[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(p).converge(described_recipe)
+      end
+      it 'converges successfully' do
+        expect { chef_run }.to_not raise_error
+      end
```
Comment (fahlmant) [inline spec/default_spec.rb:11]: Since it's a small, one recipe cookbook, adding all the tests here may be a good idea. Thoughts @ramereth ?

**Review by fahlmant — DISMISSED** [review]

File: spec/default_spec.rb (hunk 2 above)
Comment (ramereth) [reply spec/default_spec.rb:11]: Let's keep that in a different PR for now.

File: .rubocop.yml (hunk 1 above)
Comment (ramereth) [reply .rubocop.yml:20]: Let's go ahead and keep it. We'll need it when we upgrade chefdk soon.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+CENTOS_6 = {
+  platform: 'centos',
+  version: '6.7'
+}.freeze
+
+DEBIAN_8 = {
+  platform: 'debian',
+  version: '8.4'
+}.freeze
```
Comment (ramereth) [inline spec/spec_helper.rb]: Please remove Debian 8.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by fahlmant — APPROVED** [review]
Once Ramereth's comments are resolved

**Review by ramereth — APPROVED** [review]

**Comment by cthalmann** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.0.1.  Have a nice day!

**Pull request merged**